src/arm/debugger/debugger.c

```diff
@@ -115,6 +126,9 @@ static bool ARMDebuggerUpdateStackTraceInternal(struct mDebuggerPlatform* d, uin
 			bool isBranch = ARMInstructionIsBranch(info.mnemonic);
 			int reg = (isBranch ? info.op1.reg : info.op2.reg);
 			destAddress = cpu->gprs[reg];
+			if ((info.operandFormat & ARM_OPERAND_MEMORY_2) && (info.branchType & ARM_BRANCH_INDIRECT)) {
+				destAddress = cpu->memory.load32(cpu, destAddress, NULL);
```

Comment:
This won't work; you'll need to parse `info.memory`, which I don't think there's a helper for. However, I should add one at some point. Though, I'm not entirely clear on if this is running before or after the instruction is executed, so it might already be in `pc` (if `info.op1.reg` is 15 that is) by the time this is called.

Reply:
With this last push, I'm about 98% sure I've done this right but in particular I can't quite tell how preincrement is supposed to work.

Reply:
Pre-increment means that the increment that's done is applied to the address before the actual access (and then written back), post-increment means the increment is done after the access.